### ISSUE TITLE
Authenticated user shows as anonymous in log entries

### ISSUE THREAD

**Issue by joeribekker** [open-formulieren/open-forms#944]
Even when a BSN is known, the log entry indicates:

```
[2021-11-16 11:02:41 CET] (Inzending XXX): Anonieme gebruiker start nieuwe inzending voor formulier "XXX" (ID: XX)
```

I expected this to show "Geauthenticeerde gebruiker..."

**Comment by sergei-maertens** [comment]
This is correct. Even if a user is authenticated with DigiD, they are _not_ authenticated in Django, which is what is checked/logged here.

From the issue description, it sounds like this functionality should be changed to look at the session if any of the _form_ authentication plugins was used rather than checking `user.is_authenticated`. Correct?

**Comment by joeribekker** [comment]
There's a difference between log entries that are caused by form-users, and some AVG entries that are caused by admin-users.

**Comment by joeribekker** [comment]
Even bespreken in call met Bart en Sergei.

**Comment by sergei-maertens** [comment]
@Bartvaderkin you raised some concerns that we haven't discussed yet, but https://github.com/open-formulieren/open-forms/issues/957#issuecomment-983699601 sketches out some changes that should facilitate this ticket.

The takeaways are:

* After authentication, on the `request.session` there will be a key `form_auth` with a value a dict:
  ```py
  {
     "plugin_id": "digid",
     "attribute": "bsn",
     "value": "123456782"
   }
  ```
* This data will also be saved on `Submission` records, if authenticated.

This means that the presence of `form_auth` session key will be sufficient, and you can enrich this with which plugin was used for the audit trail. For downstream actions, this information can also be derived from the `Submission` record.

The data in there is considered immutable - if the user resumes the form, they have to use the same authentication method.
